utils/common/src/main/java/org/apache/brooklyn/util/http/HttpTool.java

```diff
@@ -250,6 +254,21 @@ public static HttpClientBuilder httpClientBuilder() {
         private boolean trustAll;
         private boolean trustSelfSigned;
 
+        public static HttpClientBuilder fromBuilder(HttpClientBuilder other) {
+            HttpClientBuilder result = httpClientBuilder();
+            result.clientConnectionManager = other.clientConnectionManager;
+            result.httpParams = other.httpParams;
```

Comment:
HttpClient should be configured with `clientConnectionManager` possibly the `HttpExecutorFactory` and other similar client properties.
A client can be used against multiple endpoints so including `uri`, `port` and `credentials` is not appropriate for the HttpClientBulder in my opinion.